### PR TITLE
member onboarding migration

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This migration will remarshal all existing members in the database:

1. Change `DateAdded` to `JoinedAt`
2. Populate organization name and domain
3. Populate default developer and professional segments so existing users do not have to do onboarding.
4. Set the `invited` flag and missing `JoinedAt` to the correct values

This does require a CSV file with the existing organizations in order to populate the org info on each member.

Fixes SC-20672

### Type of change

- [ ] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [x] migration

### Acceptance criteria

[Copy acceptance criteria checklist from story for reviewer, or add a brief acceptance criteria here]

[Front-End: add screenshots/videos of changes made]

### Definition of Done

- [ ] I have manually tested the change running it locally (having rebuilt all containers) or via unit tests 
- [ ] I have added unit and/or integration tests that cover my changes
- [ ] I have added new test fixtures as needed to support added tests
- [ ] I have updated the dependencies list if necessary (including updating yarn.lock and/or go.sum)
- [ ] I have recompiled and included new protocol buffers to reflect changes I made if necessary
- [ ] Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ] I have notified the reviewer via Shortcut or Slack that this is ready for review
- [ ] Front-end: Checked sm, md, lg screen resolutions for effective responsiveness
- [ ] Backend-end: Documented service configuration changes or created related devops stories

### Reviewer(s) checklist

- [ ] Front-end: I've reviewed the Figma design and confirmed that changes match the spec.
- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

**Review by bbengfort — APPROVED** [review]
Ok, this looks good - I didn't review in detail since this is migration code which I assume won't be merged, but in broad strokes I think I understand what is happening.

So we'll need to output a CSV from quarterdeck with the ID,Name,Domain of all organizations in order to run this migration? Is there a story for creating that CSV output or does that already exist in quarterdeck?

File: cmd/tenant/main.go [hunk 1]
```diff
@@ -149,6 +179,218 @@ func usage(c *cli.Context) (err error) {
...
+		} else {
+			if err = db.UpdateMember(ctx, member); err != nil {
+				errs = multierror.Append(errs, err)
+				if reportFile != nil {
+					reportFile.WriteString(fmt.Sprintf("error migrating member %s:%s: %s\n", old.OrgID.String(), old.ID.String(), err.Error()))
+				}
+				return nil
+			}
+
+			fmt.Println(fmt.Sprintf("migrated record: %v", member.ToAPI()))
```
Comment (bbengfort) [inline cmd/tenant/main.go:366]:
```suggestion
			fmt.Printf("migrated record: %v\n", member.ToAPI())
```

Comment (bbengfort) [reply cmd/tenant/main.go:366]: to appease the linter

File: cmd/tenant/main.go [hunk 2]
```diff
@@ -149,6 +179,218 @@ func usage(c *cli.Context) (err error) {
...
+
+	// Read the header
+	var header []string
+	reader := csv.NewReader(orgsFile)
+	if header, err = reader.Read(); err != nil {
+		return cli.Exit(err, 1)
+	}
+
+	// Ensure header fields are correct
+	if header[0] != "ID" || header[1] != "Name" || header[2] != "Domain" {
```
Comment (bbengfort) [inline cmd/tenant/main.go:226]: Unless the CSV already export exists, I would usually name the headers the same name as the columns in the database, which are lower snake case. 

```suggestion
	if header[0] != "id" || header[1] != "name" || header[2] != "domain" {
```

Comment (pdeziel) [reply cmd/tenant/main.go:226]: I already have the CSV export from running SC-21250, this matches the header format defined there.

**Pull request merged**